magefiles/main.go

```diff
@@ -14,27 +15,101 @@ import (
 
 // BootstrapTools installs all tools needed tobuild and release Armada.
 // For the list of tools this will install, see tools.yaml in the root directory
+type ToolsList struct {
+	Tools []string
+}
+
 func BootstrapTools() error {
 	mg.Deps(goCheck)
 	type ToolsList struct {
 		Tools []string
 	}
 
-	tools := &ToolsList{}
-	err := readYaml("tools.yaml", tools)
+	if isGitHubActions() {
+		gomodcache, err := os.MkdirTemp("", "gomodcache")
+		if err != nil {
+			fmt.Println("Error creating temporary GOMODCACHE directory:", err)
+			return err
+		}
+		defer os.RemoveAll(gomodcache)
+
+		gocache, err := os.MkdirTemp("", "gocache")
+		if err != nil {
+			fmt.Println("Error creating temporary GOCACHE directory:", err)
+			return err
+		}
+		defer os.RemoveAll(gocache)
+
+		os.Setenv("GOMODCACHE", gomodcache)
+		os.Setenv("GOCACHE", gocache)
+	}
+
+	armadaToolsFilePath, err := getArmadaToolsFilePath()
 	if err != nil {
 		return err
 	}
 
-	for _, tool := range tools.Tools {
-		err := goRun("install", tool)
-		if err != nil {
+	if _, err := os.Stat(armadaToolsFilePath); os.IsNotExist(err) {
+		if err := createEmptyToolsList(armadaToolsFilePath); err != nil {
```

Comment:
why do you create an empty tools list? I think it may be better to try to read from the tools list and make it an empty list when the file does not exist, e.g.

```go
	existingTools := &ToolsList{}
 	if err := readYaml(armadaToolsFilePath, existingTools); err != nil && !os.IsNotExist(err) {
 		return err
 	}
```